src/libraries/Microsoft.PowerFx.Core/Utils/TypeUtils.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Linq;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Core.Utils
+{
+    internal static class TypeUtils
```

Comment:
>TypeUtils [](http://example.com/codeflow?start=26&length=9)

This is only needed in StructuralPrint, so there's no need to make it its own class - it can be a "normal" private method in StructuralPrint.

Reply:
Nevermind, saw that it's used in the interpreter (SetFunction). Any reason why it can't be made part of DType itself?

Reply:
No particular reason. This was previously implemented in the interpreter and then I had to move it to core.

Reply:
In the interpreter it made sense to be an extension method - we didn't want to "pollute"  the core project with something that would not be used there. But now that it is, this can be moved to the DType class itself instead of adding a new utility class (we already have quite a few of them)